### ISSUE TITLE
padding-line-between-statements doesn't treat IIFEs as functions

### ISSUE THREAD

**Issue by LJNeon** [eslint/eslint#10843]
**Tell us about your environment**

* **ESLint Version:** 5.5.0
* **Node Version:** 10.9.0
* **npm Version:** 6.2.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yml
env: {
  es6: true,
  node: true
}

extends: "eslint:recommended"

rules: {
  padding-line-between-statements: [2, {
    blankLine: "never",
    next: "*",
    prev: "*"
  },
  {
    blankLine: "always",
    next: "function",
    prev: "*"
  },
  {
    blankLine: "always",
    next: "*",
    prev: "function"
  }]
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
process.on("unhandledRejection", (r, p) => {
  // ...
});
// <---- eslint wants me to remove this blank line
(async () => {
  // ...
})();
```

```bash
eslint . --ext .js
```

**What did you expect to happen?**
It would consider the code valid as I've stated that function should have blank lines both before and after.
**What actually happened? Please include the actual, raw output from ESLint.**
```bash
/../src/index.js
  34:1  error  Unexpected blank line before this statement  padding-line-between-statements
```

**Comment by platinumazure** [comment]
Hi @LJNeon, thanks for the issue.

It looks like IIFEs are covered under the `"block-like"` or `"multiline-block-like"` statement types of this rule. Those statement types also cover standalone blocks, function declarations/expressions, and if/while/do-while/switch/for statements that have blocks, so they may be too broad for your use case.

I could see a case for enhancing the rule with a new `"iife"` statement type, which would only cover IIFEs. Then you could use `["function", "iife"]` as the statement type in your rule config.

What do you think?

**Comment by LJNeon** [comment]
That sounds good to me, I just need a way to treat iifes as functions.

**Comment by platinumazure** [comment]
@LJNeon I've created #10853 to track the enhancement request. Please review and let me know if you think it needs any changes.

If all looks well and you're okay with tracking that issue, please feel free to close this. Thanks!

**Comment by LJNeon** [comment]
Looks good to me
